prExplorer/client-app/src/app/stores/productStore.ts

```diff
@@ -0,0 +1,115 @@
+import { makeAutoObservable, runInAction } from "mobx";
+import { Product } from "../models/product";
+import agent from "../api/agent";
+import { v4 as uuid } from "uuid";
+
+export default class ProductStore {
+  productRegistry = new Map<string, Product>();
```

Comment:
Why is it a `Map`? Won't an array be sufficient?

Reply:
never worked with js objects before. thats why i wanted to try it.

Reply:
ok ;)